pkg/sql/opt/table_meta.go

```diff
@@ -254,7 +255,10 @@ func (tm *TableMeta) IsIndexNotVisible(indexOrd cat.IndexOrdinal, rng *rand.Rand
 			isNotVisible = true
 		}
 	}
-	tm.notVisibleIndexMap[indexOrd] = isNotVisible
+	if isNotVisible {
+		tm.cachedIndexVisibility.Add(indexOrd)
```

Comment:
Don't we want to set `tm.cachedIndexVisibility` even if the index is visible to avoid re-rolling the dice next time `IsIndexNotVisible` is called?